packages/gen-ai-hub/src/orchestration/orchestration-client.ts

```diff
@@ -1,34 +1,42 @@
 import { executeRequest, CustomRequestConfig } from '@sap-ai-sdk/core';
-import { CompletionPostRequest } from './client/api/schema/index.js';
 import {
-  GenAiHubCompletionParameters,
-  GenAiHubCompletionResponse
-} from './orchestration-types.js';
+  DeploymentResolver,
+  resolveDeployment
+} from '../utils/deployment-resolver.js';
+import {
+  CompletionPostRequest,
+  CompletionPostResponse
+} from './client/api/schema/index.js';
+import { OrchestrationCompletionParameters } from './orchestration-types.js';
 
 /**
  * Get the orchestration client.
  */
-export class GenAiHubClient {
+export class OrchestrationClient {
   /**
    * Creates a completion for the chat messages.
    * @param data - The input parameters for the chat completion.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - Request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
-    data: GenAiHubCompletionParameters,
+    data: OrchestrationCompletionParameters,
+    deploymentResolver: DeploymentResolver = () =>
+      resolveDeployment({ scenarioId: 'orchestration' }),
     requestConfig?: CustomRequestConfig
-  ): Promise<GenAiHubCompletionResponse> {
-    const dataWithInputParams = {
-      deploymentConfiguration: data.deploymentConfiguration,
-      ...constructCompletionPostRequest(data)
-    };
+  ): Promise<CompletionPostResponse> {
+    const body = constructCompletionPostRequest(data);
+    const deployment =
+      typeof deploymentResolver === 'function'
+        ? (await deploymentResolver()).id
+        : deploymentResolver;
```

Comment:
Regarding my comment before (OpenAIClient): Maybe this can even be extracted to core, so that it is not repeated even more. I think it is fine to duplicate it here due to the package boundaries (I mean it's 4 loc), however once we add more model-specific clients we will need to solve it somehow anyways.